[PRD-5963] - Date Picker showing One Day Prior for Default Value of =…

cravobranco opened the pull request:
…TODAY and =YESTERDAY when client is in Different Timezone

   * Added client side support to handle reports which date is calculated based on client timezone
   * Refactored the date parser based on timezone type value, which can be one of the following:
    1) server - timezone already handled by the server
    2) client - timezone handled in the client side (see pentaho-platform-plugin-common-ui - formatting.js)
    3) utc - timezone forced to UTC
    4) customized timezone - timezone forced to customized timezone

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'core' -P '!assemblies' -Dsurefire.runOrder=alphabetical -Daudit -amd clean install
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java
  core/src/main/javascript/reportviewer/reportviewer-formatting.js
  core/src/test/java/org/pentaho/reporting/platform/plugin/ReportContentUtilTest.java
```

#### License header violations

> :heavy_exclamation_mark: Copyright year is not the current year on core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java. Found 2017. Was expecting 2018

> :heavy_exclamation_mark: Couldn't find acceptable copyright on core/src/test/java/org/pentaho/reporting/platform/plugin/ReportContentUtilTest.java

> :heavy_exclamation_mark: Couldn't find acceptable license on core/src/test/java/org/pentaho/reporting/platform/plugin/ReportContentUtilTest.java

wingman-pentaho commented:
### Build Completed

:white_check_mark: This pull request has passed all validations.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'core' -P '!assemblies' -Dsurefire.runOrder=alphabetical -Daudit -amd clean install
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java
  core/src/main/javascript/reportviewer/reportviewer-formatting.js
  core/src/test/java/org/pentaho/reporting/platform/plugin/ReportContentUtilTest.java
```

cravobranco commented:
@pentaho-lmartins , @pamval , @mbatchelor , @mchen-len-son , @tmorgner  please review and merge if it's all ok

Notes: This should be reviewed together with:
https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1296

Main changes:
- Added client side support to handle reports which date is calculated based on client timezone
- Refactored the date parser based on timezone type value, which can be one of the following:
1) server - timezone already handled by the server
2) client - timezone handled in the client side (see pentaho-platform-plugin-common-ui - formatting.js)
3) utc - timezone forced to UTC
4) customized timezone - timezone forced to customized timezone

wingman-pentaho commented:
### Build Completed

:white_check_mark: This pull request has passed all validations.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'core' -P '!assemblies' -Dsurefire.runOrder=alphabetical -Daudit -amd clean install
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'core' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  core/src/main/java/org/pentaho/reporting/platform/plugin/ReportContentUtil.java
  core/src/main/javascript/reportviewer/reportviewer-formatting.js
  core/src/test/java/org/pentaho/reporting/platform/plugin/ReportContentUtilTest.java
```

cravobranco commented:
This case is now handled in #636
